js/base/Exchange.js

```diff
@@ -1466,6 +1466,56 @@ module.exports = class Exchange {
         return result;
     }
 
+    buildOHLCVC (trades, timeframe = '1m', since = undefined, limit = undefined) {
+        if (since === undefined) {
+            since = 0;
+        }
+        if (limit === undefined) {
+            limit = 2147483647; // max int32
+        }
+        // given a sorted arrays of trades (recent last) and a timeframe builds an array of OHLCV candles
+        const ms = this.parseTimeframe (timeframe) * 1000;
+        const ohlcvs = [];
+        const i_timestamp = 0;
+        // const open = 1;
+        const i_high = 2;
+        const i_low = 3;
+        const i_close = 4;
+        const i_volume = 5;
+        const i_count = 6;
+        const tradesLength = trades.length;
+        const oldest = Math.min (tradesLength, limit);
+        for (let i = 0; i < oldest; i++) {
+            const trade = trades[i];
+            if (trade['timestamp'] < since) {
+                continue;
+            }
+            const openingTime = Math.floor (trade['timestamp'] / ms) * ms; // shift to the edge of m/h/d (but not M)
+            const ohlcv_length = ohlcvs.length;
+            const candle = ohlcv_length - 1;
+            if (candle === -1 || openingTime >= ohlcvs[candle][i_timestamp] + ms) {
```

Comment:
We can't use operator `+`, we must use `this.sum()` here instead: 

```suggestion
            if ((candle === -1) || (openingTime >= this.sum (ohlcvs[candle][i_timestamp], ms))) {
```